File: src/tests/stuff/moretest/collatz.go
package main

import "fmt"

func main() {
	fmt.Println("collatz")
	total := 0
	for k := 1; k <= 50000; k++ {
		n := k
		steps := 0
		for n > 1 {
			if n%2 == 0 {
				n = n / 2
			} else {
				n = 3*n + 1
			}
			steps++
		}
		total += steps
	}
	fmt.Println(total)
}

File: src/tests/stuff/moretest/collatz.rs
fn main() {
    println!("collatz");
    let mut total: i64 = 0;
    for k in 1..=50000 {
        let mut n = k;
        let mut steps: i64 = 0;
        while n > 1 {
            if n % 2 == 0 { n /= 2; }
            else { n = 3 * n + 1; }
            steps += 1;
        }
        total += steps;
    }
    println!("{}", total);
}
